p2p/pex/addrbook.go

```diff
@@ -55,7 +55,7 @@ type AddrBook interface {
 	PickAddress(biasTowardsNewAddrs int) *p2p.NetAddress
 
 	// Mark address
-	MarkGood(*p2p.NetAddress)
+	MarkGood(p2p.ID)
 	MarkAttempt(*p2p.NetAddress)
```

Comment:
Should we refactor MarkAttempt and MarkBad to accept p2p.ID instead of NetAddress? Otherwise interface is inconsistent.

Reply:
Looks like we can't do that because `MarkBad` calls `RemoveAddress`, which requires `NetAddress`.